src/partr.c

```diff
@@ -689,7 +689,7 @@ JL_DLLEXPORT jl_task_t *jl_task_new(jl_value_t *_args)
 {
     jl_ptls_t ptls = jl_get_ptls_states();
 
-    jl_task_t *task = (jl_task_t *)jl_new_struct_uninit(jl_task_type);
+    jl_task_t *task = (jl_task_t *)jl_gc_alloc(ptls, sizeof (jl_task_t), jl_task_type);
```

Comment:
Why are you using `jl_gc_alloc` directly? 

Is there a size difference between `jl_datatype_size(jl_task_type)`  and `sizeof(jl_task_t)`?

Reply:
Sadly, yes. `jl_task_t` has a bunch of hidden fields.